### PR TITLE
Enable ORCA on macOS on Travis, and assorted build improvements

### PR DESCRIPTION
**Pull request by d**
We disabled ORCA in the macOS Travis build in commit 649ee57dda3af
(#10068) when we thought builds timed out because the build machines
were underpowered. Turns out we made a mistake in 649ee57dda3af where
the mac build was just broken (see fix in commit b371e59227dde #10147).
Now that all the pieces are in place, I can think of no reason to
neglect this compile-time coverage in Travis.

Given that the macOS VMs in Travis are generally underpowered, build
with -O0 to expedite the process.

Also included in the patch set are two smaller changes:

* Update `AX_CXX_COMPILE_STDCXX` to latest version
* Drop `-Wno-variadic-macros`, which is inapplicable.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by d** [comment]
Note to reviewers: the "drop me" commit is not meant for merging, and will be dropped once we get some feedback on this patch set.

File: src/backend/gporca/gporca.mk [hunk 1]
```diff
@@ -4,4 +4,4 @@ override CPPFLAGS := -I$(top_srcdir)/src/backend/gporca/libnaucrates/include $(C
 override CPPFLAGS := -I$(top_srcdir)/src/backend/gporca/libgpdbcost/include $(CPPFLAGS)
 # Do not omit frame pointer. Even with RELEASE builds, it is used for
 # backtracing.
-override CXXFLAGS := -Werror -Wextra -Wpedantic -Wno-variadic-macros -fno-omit-frame-pointer $(CXXFLAGS)
```
Comment (chrishajas) [inline src/backend/gporca/gporca.mk:7]: Should this also be removed from `src/backend/gporca/CMakeLists.txt` since we require C++ 14 there too?

Comment (d) [reply src/backend/gporca/gporca.mk:7]: oops yes

**Comment by d** [comment]
@ashuka24 and @chrishajas I just amended the commit that removes the warning suppression for variadic macros to take into account our `CMakeLists.txt`. Also dropping the "drop me" commit to get this closer to merging, wanna take one last quick look?

The commit history as is is ready for merging.

**Pull request merged**